test/org/sagebionetworks/bridge/services/SurveyServiceMockTest.java

```diff
@@ -116,11 +133,46 @@ public void successfulDeletePermanently() {
         Survey survey = createSurvey();
         
         service.deleteSurveyPermanently(TEST_STUDY, survey);
-        
+
+        // verify query args
+        ArgumentCaptor<String> queryCaptor = ArgumentCaptor.forClass(String.class);
+        verify(mockSharedModuleMetadataService).queryAllMetadata(anyBoolean(), anyBoolean(), queryCaptor.capture(),  anySetOf(String.class));
+
+        String queryStr = queryCaptor.getValue();
+        assertEquals(queryStr, "surveyGuid=\'" + survey.getGuid() + "\' AND surveyCreatedOn=" + survey.getCreatedOn());
+
         verify(mockSurveyDao).deleteSurveyPermanently(keysCaptor.capture());
         assertEquals(survey, keysCaptor.getValue());
     }
-    
+
+    @Test(expected = BadRequestException.class)
+    public void logicallyDeleteSurveyNotEmptySharedModules() {
+        when(mockSharedModuleMetadataService.queryAllMetadata(anyBoolean(), anyBoolean(), anyString(), anySetOf(String.class)))
+                .thenReturn(ImmutableList.of(makeValidMetadata()));
+
+        Survey survey = createSurvey();
+        doReturn(survey).when(mockSurveyDao).getSurvey(any());
+        service.deleteSurvey(survey);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void physicallyDeleteSurveyNotEmptySharedModules() {
+        when(mockSharedModuleMetadataService.queryAllMetadata(anyBoolean(), anyBoolean(), anyString(), anySetOf(String.class)))
+                .thenReturn(ImmutableList.of(makeValidMetadata()));
+
+        doReturn(ImmutableList.of()).when(mockSchedulePlanService).getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY);
+        Survey survey = createSurvey();
+
+        service.deleteSurveyPermanently(TEST_STUDY, survey);
+
+        // verify query args
+        ArgumentCaptor<String> queryCaptor = ArgumentCaptor.forClass(String.class);
+        verify(mockSharedModuleMetadataService).queryAllMetadata(eq(false), eq(false), queryCaptor.capture(), eq(null));
+
+        String queryStr = queryCaptor.getValue();
+        assertEquals(queryStr, "surveyGuid=\'" + survey.getGuid() + "\' AND surveyCreatedOn=" + survey.getCreatedOn());
```

Comment:
This test throws, so anything after the deleteSurveyPermanently() won't be executed.